cblite/llm/Gemini.cc

```diff
@@ -23,32 +23,20 @@ using namespace fleece;
 using namespace litecore;
 
 vector<alloc_slice> Gemini::run(const string& modelName, vector<Value> propVec) {
-    int batchSize = 50, docRemainder = propVec.size() % batchSize;
-    long batches = propVec.size() / batchSize;
-    if (propVec.size() % batchSize != 0) {
-        batches++;
-    }
+    const int BATCH_SIZE = 50;
+    int docRemainder = propVec.size() % BATCH_SIZE;
+    int batches = (int)((propVec.size() + BATCH_SIZE - 1) / BATCH_SIZE);
     vector<alloc_slice> responses;
     alloc_slice response, completeResponse;
     Value rawSrcPropValue;
     
     for (int i = 0; i < batches; i++) {
         // Create rest body
         string restBody = format("{\"requests\": [");
-        if ((i == batches - 1) && (docRemainder != 0)) {
-            int j = 0;
-            while (j < docRemainder) {
-                rawSrcPropValue = propVec.at(j);
-                restBody += format(" {\"model\": \"models/%s\", \"content\": {\"parts\": [{\"text\": \"%.*s\"}]}, },", modelName.c_str(), SPLAT(rawSrcPropValue.asString()));
-                j++;
-            }
-            propVec.erase(propVec.begin(), propVec.begin() + (docRemainder - 1));
-        } else {
-            for (int k = 0; k < batchSize; k++) {
-                rawSrcPropValue = propVec.at(k);
-                restBody += format(" {\"model\": \"models/%s\", \"content\": {\"parts\": [{\"text\": \"%.*s\"}]}, },", modelName.c_str(), SPLAT(rawSrcPropValue.asString()));
-            }
-            propVec.erase(propVec.begin(), propVec.begin() + (batchSize - 1));
+        int limit = ((i == batches - 1) && (docRemainder != 0)) ? docRemainder : BATCH_SIZE;
```

Comment:
This is an incorrect condition.  docRemainder != 0 is not relevant.  In fact, this will cause harm because if it IS zero, it will continue using BATCH_SIZE incorrectly.

Reply:
If I get rid of `docRemainder != 0` and the docRemainder is 0 then it will set the limit to 0

Reply:
That's not really an issue.  The whole loop will be skipped in that case and no harm.